Content.Server/Power/EntitySystems/BatterySystem.cs

```diff
@@ -103,14 +90,27 @@ private void OnEmpPulse(EntityUid uid, BatteryComponent component, ref EmpPulseE
             UseCharge(uid, args.EnergyConsumption, component);
         }
 
+        private void DirtyWithRateLimit(EntityUid uid, BatteryComponent battery)
+        {
+            // Limit network state update rate
+            if (_timing.CurTime > battery.NextSyncTime)
+            {
+                Dirty(uid, battery);
+                battery.NextSyncTime = _timing.CurTime + _syncDelay;
+            }
+        }
+
```

Comment:
This is going to be very janky ingame. The problem is if the battery gets dirtied in between the next allowed time but then never after it's always going to mispredict on the client while in range and never get back in sync.

Doing this properly would require syncing what the rate of change is to the client instead of mispredicting constantly like this.